ChainRouter.php

```diff
@@ -291,8 +291,19 @@ private function rebuildRequest($uri)
                 $server['HTTP_HOST'] .= ':'.$this->context->getHttpPort();
             }
         }
+        // use the array field to have both domain and port.
+        if (isset($server['HTTP_HOST'])) {
+            $pathinfo = $server['HTTP_HOST'].$pathinfo;
+        } else {
+            $pathinfo = 'localhost'.$pathinfo;
```

Comment:
I am not deep enough into this PR to really tell either .. @tobion might know best in general as he knows the entire routing topic really well ..